napari/_qt/_qapp_model/qactions/__init__.py

```diff
@@ -0,0 +1,52 @@
+from functools import lru_cache
+from itertools import chain
+from typing import Optional
+
+# Submodules should be able to import from most modules, so to
+# avoid circular imports, don't import submodules at the top level here,
+# import them inside the init_qactions function.
+
+
+@lru_cache  # only call once
+def init_qactions() -> None:
+    """Initialize all Qt-based Actions with app-model
+
+    This function will be called in _QtMainWindow.__init__().  It should only
+    be called once (hence the lru_cache decorator).
+
+    It is responsible for:
+    - injecting Qt-specific names into the application injection_store namespace
+      (this is what allows functions to be declared with annotations like
+      `def foo(window: Window)` or `def foo(qt_viewer: QtViewer)`)
+    - registering provider functions for the names added to the namespace
+    - registering Qt-dependent actions with app-model (i.e. Q_* actions).
+    """
+
+    from ...._app_model import get_app
+    from ...qt_main_window import Window, _QtMainWindow
+    from ...qt_viewer import QtViewer
+    from ._view import Q_VIEW_ACTIONS
+
+    # update the namespace with the Qt-specific types/providers/processors
+    app = get_app()
+    store = app.injection_store
+    store.namespace = {
+        **store.namespace,
+        'Window': Window,
+        'QtViewer': QtViewer,
+    }
+
+    # Qt-specific providers/processors
+    @store.register_provider
+    def _provide_window() -> Optional[Window]:
```

Comment:
See: https://github.com/napari/napari/pull/4977#discussion_r972901290

Reply:
Could we test this by creating a `viewer` and registering a function that requires a window, and asserting that the returned window is the viewer's window? It's indirect but it's all I can think of 🤷

Reply:
It's always best to add coverage when we can but this is an inner function whose behavior is largely constrained by the type. Do we have a test that the injection store has a `Window` under qt?